pantos-io/ci-workflows
# [PAN-1866] add python shared CI

## What type of PR is this? (check all applicable)


## Description

- Implement Pantos Python CI shared pipeline
- Split the actions which install Poetry from the action which install the dependency of the project: this is done because sometimes we need **only** poetry to build wheels and we should reuse this step and not redefine it in individual projects.
- Removed some inputs from the previous version: They were actually redundant and we can achieve the same behaviour without them.
- Compared to the previous shared pipeline that we had, I split the SonarCloud step from the unit test one. They should be independent from my point of view, especially because this step will fail if the pipeline is triggered from forks, and it should not have an impact on the unit test step. This was accomplished by passing the coverage artifacts to the SonarCloud responsible job.
- Tagged it to v1 and v1.0.0 according to https://github.com/actions/toolkit/blob/main/docs/action-versioning.md

## QA Instructions

- Pantos Python projects should be able to call the reusable workflow to check the desired properties for CI
- Pantos Python projects should be able to upload artifacts to SonarCloud using the workflow (only after the CI is successfully finished)
